rust-src/concordium_base/src/lib.rs

```diff
@@ -32,6 +32,8 @@ pub mod ps_sig;
 
 pub mod dodis_yampolskiy_prf;
 
+pub use ed25519_dalek::{PublicKey, SecretKey, Signature};
```

Comment:
I don't think we should have these exported at the top level of the crate, since we will now export both a `PublicKey` and a `VerifyKey` where the type `VerifyKey` is what we want people to use.
Also, `SignatureEd25519` and this `Signature` are sort of 'overlapping' in meaning.

However, it does make sense to expose these as they are a peer dependency, so maybe do:
```suggestion
pub use ed25519_dalek;
```

Reply:
There was a discussion in https://github.com/Concordium/concordium-base/pull/477 to export these three types.

Reply:
I don't want to expose the entire dalek crate here. Exposing third party crates complicates our updates and versioning of our own crate. I prefer to expose types that appear in our API but not everything else.

But it's a good point that exposing these at the top-level under these names is not ideal, so we should rename them.

Perhaps
```rust
pub mod ed25519 {
          pub use ed25519_dalek::{PublicKey, SecretKey, Signature};
}
```
with some documentation would be most understandable.

Reply:
I added an explanation and changed it:

```
/// We expose the `PublicKey`, `SecretKey`, and `Signature` from the third-party `ed25519_dalek` crate here because these types appear in Concordium's API.
pub mod ed25519 {
    pub use ed25519_dalek::{PublicKey, SecretKey, Signature};
}
```